vyper/venom/basicblock.py

```diff
@@ -237,6 +241,34 @@ def __repr__(self):
         return json.dumps(self.value)  # escape it
 
 
+@dataclass(frozen=True)
+class MemoryLocation:
+    """Represents a memory location that can be analyzed for aliasing"""
+
+    offset: int = 0
+    size: int = 0
+    is_volatile: bool = False
+
+    def completely_overlaps(self, other: "MemoryLocation") -> bool:
```

Comment:
since `overlaps` is kind of a "commutative" word (like from the name i would expect `self.completely_overlaps(other) == other.completely_overlaps(self)`), maybe we should call this `completely_contains`?